time-benchmarks/benches/offset_date_time.rs

```diff
@@ -0,0 +1,660 @@
+        ben.iter(|| (
+            a.millisecond(),
+            b.millisecond(),
+        ));
+    }
+
+    microsecond(ben: &mut Bencher) {
+        let a = datetime!("2019-01-01 0:00 UTC");
+        let b = datetime!("2019-01-01 23:59:59.999_999 UTC");
+        ben.iter(|| (
+            a.microsecond(),
+            b.microsecond(),
+        ));
+    }
+
+    nanosecond(ben: &mut Bencher) {
+        let a = datetime!("2019-01-01 0:00 UTC");
+        let b = datetime!("2019-01-01 23:59:59.999_999_999 UTC");
+        ben.iter(|| (
+            a.nanosecond(),
+            b.nanosecond(),
+        ));
+    }
+
+    format(ben: &mut Bencher) {
+        let a = datetime!("2019-01-02 0:00 UTC");
+        let b = datetime!("2019-01-02 3:04:05.678_901_234").assume_offset(offset!("+6:07"));
+        ben.iter(|| (
+            a.format("%F %r %z"),
+            b.format(Format::Rfc3339),
+        ));
+    }
+
+    parse(b: &mut Bencher) {
+        b.iter(|| (
+            OffsetDateTime::parse("2019-01-02 00:00:00 +0000", "%F %T %z"),
+            OffsetDateTime::parse("2019-002 23:59:59 +0000", "%Y-%j %T %z"),
+            OffsetDateTime::parse("2019-W01-3 12:00:00 pm +0000", "%G-W%V-%u %r %z"),
+            OffsetDateTime::parse("2019-01-02 03:04:05 +0600", "%F %T %z"),
+            OffsetDateTime::parse("2020-09-08T08:44:31+02:30", Format::Rfc3339),
+            OffsetDateTime::parse("2019-01-02T03:04:05.678901234+05:06", Format::Rfc3339),
+            OffsetDateTime::parse("2019-01-02T03:04:05.678901234Z", Format::Rfc3339),
+            OffsetDateTime::parse("2019-01-02T03:04:05/", Format::Rfc3339),
+            OffsetDateTime::parse("2019-01-02T03:04:05", Format::Rfc3339),
+            OffsetDateTime::parse("2019-01-02T03:04:05.", Format::Rfc3339),
+        ));
+    }
+
+    partial_eq(ben: &mut Bencher) {
+        let a = datetime!("2000-01-01 0:00 UTC").to_offset(offset!("-1"));
+        let b = datetime!("2000-01-01 0:00 UTC");
+        ben.iter(|| a == b);
+    }
+
+    partial_ord(ben: &mut Bencher) {
+        let a = datetime!("2019-01-01 0:00 UTC");
+        let b = datetime!("2019-01-01 0:00 UTC").to_offset(offset!("-1"));
+        ben.iter(|| a.partial_cmp(&b));
+    }
+
+    ord(ben: &mut Bencher) {
+        let a = datetime!("2019-01-01 0:00 UTC");
+        let b = datetime!("2019-01-01 0:00 UTC").to_offset(offset!("-1"));
+        let c = datetime!("2019-01-01 0:00:00.000_000_001 UTC");
+        ben.iter(|| (a == b, c > a));
+    }
+
+    hash(ben: &mut Bencher) {
+        use std::collections::hash_map::DefaultHasher;
+        use std::hash::{Hash, Hasher};
+
+        let a = datetime!("2019-01-01 0:00 UTC");
+        let b = datetime!("2019-01-01 0:00 UTC").to_offset(offset!("-1"));
+        let c = datetime!("2019-01-01 0:00");
+
+        ben.iter(|| (
+            {
+                let mut hasher = DefaultHasher::new();
+                a.hash(&mut hasher);
+                hasher.finish()
```

Comment:
I believe only the middle line here actually needs to be in the loop. I may be wrong on this one, though.

Reply:
Hm, yeah, I guess so. I was thinking this checks the whole hashing procedure from start to finish, but I guess only the `hash()` call is really relevant to the performance of hashing this type specifically. That means we can even reuse the same `Hasher` instance between iteration loops.